### PR TITLE
Attachments Mag bug fix

### PR DESCRIPTION
**Pull request by kbatbouta**
## Changes

- Fixed issues related to the magazine count being larger than the magazine size due to a removed magazine attachment.
- Now pawns will unload all guns before making any changes to them.
- Added new consistency checks on save loading.

### CODE REVIEW

File: Source/CombatExtended/CombatExtended/Comps/CompAmmoUser.cs [hunk 1]
```diff
@@ -57,15 +57,15 @@ public int MagsLeft
         public int MagSize
         {
             get
-            {
+            {
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Comps/CompAmmoUser.cs:60]: Redundant space is redundant.

File: Source/CombatExtended/CombatExtended/Comps/CompAmmoUser.cs [hunk 2]
```diff
@@ -57,15 +57,15 @@ public int MagsLeft
...
+            {                              
                 return (int)parent.GetStatValue(CE_StatDefOf.MagazineCapacity);
             }
         }
 
         public int CurMagCount
         {
             get
-            {
+            {
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Comps/CompAmmoUser.cs:68]: Similarly

File: Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs [hunk 3]
```diff
@@ -89,12 +89,18 @@ public override IEnumerable<Toil> MakeNewToils()
             Toil waitToil = new Toil();
             waitToil.defaultDuration = 30;
             waitToil.defaultCompleteMode = ToilCompleteMode.Delay;
-            yield return waitToil.WithProgressBarToilDelay(TargetIndex.A);
+            yield return waitToil.WithProgressBarToilDelay(TargetIndex.A);
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs:92]: Similarly

File: Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs [hunk 4]
```diff
@@ -89,12 +89,18 @@ public override IEnumerable<Toil> MakeNewToils()
...
             Toil modifyToil = new Toil();
             modifyToil.defaultCompleteMode = ToilCompleteMode.Instant;            
             modifyToil.initAction = () =>
-            {                
+            {
+                CompAmmoUser compAmmo = Weapon.TryGetComp<CompAmmoUser>();
+                if (compAmmo != null && compAmmo.HasMagazine && compAmmo.UseAmmo && !compAmmo.MagazineEmpty && !compAmmo.TryUnload(out _, forceUnload: true, pawn.Position))
+                {
+                    Log.Warning("CE: Failed to unload ammo from weapon before modifying it");
+                    Log.TryOpenLogWindow();
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs:102]: Leftover from debugging?

File: Source/CombatExtended/CombatExtended/Things/WeaponPlatform.cs [hunk 5]
```diff
@@ -332,7 +347,7 @@ public void UpdateConfiguration()
             /*
              * <========= update ammo stuff =========> 
              */
-            verbManager.InitializeAmmoManagers();
+            verbManager.InitializeAmmoManagers();
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Things/WeaponPlatform.cs:350]: Similarly

File: Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs (hunk 4 above)
Comment (kbatbouta) [reply Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs:102]: No, this one should be brought to the user's attention. It's not an error but it's not good either.

**Pull request merged**